openforcefield/utils/collections.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python
+
+"""
+Custom collections classes.
+
+"""
+
+__all__ = [
+    'ValidatedList',
+]
+
+
+# =====================================================================
+# GLOBAL IMPORTS
+# =====================================================================
+
+from collections import abc
+
+
+# =====================================================================
+# VALIDATED LIST
+# =====================================================================
+
+class ValidatedList(list):
```

Comment:
Aha! Ok, now I understand what you were talking about regarding the shortcomings of `Callbackable`. Yes, this is the right way to go forward for now.